File: tmdb-core/src/main/java/dev/reuss/tmdb/value/id/TvSeasonId.java
package dev.reuss.tmdb.value.id;

/**
 * TMDB TV season id.
 *
 * @param value TMDB TV season id
 */
public record TvSeasonId(int value) implements NumericTmdbResourceId {

    public TvSeasonId {
        if (value <= 0) {
            throw new IllegalArgumentException("TV season id must be greater than 0");
        }
    }

    public static TvSeasonId of(int value) {
        return new TvSeasonId(value);
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/value/id/TvSeasonId.kt
package dev.reuss.tmdb.value.id

/**
 * TMDB TV season id.
 */
class TvSeasonId private constructor(
    value: Int,
) : NumericTmdbResourceId(value, "TV season id") {
    companion object {
        /**
         * Creates an identifier from a positive TMDB resource ID.
         *
         * @throws IllegalArgumentException if [value] is not greater than `0`
         */
        @JvmStatic
        fun of(value: Int): TvSeasonId = TvSeasonId(value)
    }
}
